coral-spark/src/main/java/com/linkedin/coral/spark/transformers/ExtractUnionFunctionTransformer.java

```diff
@@ -30,14 +39,24 @@
  * See {@link CoalesceStructUtility#COALESCE_STRUCT_FUNCTION_RETURN_STRATEGY} and its comments for more details.
  *
  * Check `CoralSparkTest#testUnionExtractUDF` for examples.
+ *
+ * Note that uniontypes holding a single need to be handled specially in Spark as there is a Spark-specific mechanism
+ * that unwraps a single uniontype (a uniontype holding only one data type) to just the single underlying data type.
+ * Reference: https://spark.apache.org/docs/latest/sql-data-sources-avro.html#supported-types-for-avro---spark-sql-conversion
+ *
+ * Check `CoralSparkTest#testUnionExtractUDFOnSingleTypeUnions` for examples.
+ *
```

Comment:
This should state that the behavior is specific to base tables. Giving an example query helps. In fact, you might write a few illustrating different expected behaviors of this transformer.

Reply:
> This should state that the behavior is specific to base tables.

I'm not sure this unwrapping behavior specific to base tables.  I created a hive base table with a single union column, then created a view that selects all on the base table. Running `desc` on both table and view in Spark yield the same schema (single union is unwrapped).

> Giving an example query helps

On line 47, the comment points to [`CoralSparkTest.testUnionExtractUDFOnSingleTypeUnions`](https://github.com/linkedin/coral/pull/507/files#diff-83e46e8495b55d1ca847275a08bdbfe14395caf725b7a7253fcf69b457303bd4R425) which has examples covering simple and complex cases of Hive queries that would induce this transformer, and also the expected behavior.

Reply:
> I'm not sure this unwrapping behavior specific to base tables. I created a hive base table with a single union column, then created a view that selects all on the base table. Running desc on both table and view in Spark yield the same schema (single union is unwrapped).

I meant explaining that the main reason for unwrapping relies in base tables. The current description is a bit vague and may not help with the thought process.

Stating examples in the comments helps vs pointing to the test cases.

Reply:
@wmoustafa Updated this piece of javadoc to include end2end reasoning + examples in comments. PTAL